icu4j/main/tests/core/src/com/ibm/icu/dev/test/number/DecimalQuantityTest.java

```diff
@@ -603,6 +605,127 @@ public void testNickelRounding() {
         }
     }
 
+    @Test
```

Comment:
Please add a test for what happens when you combine `.scale()` with `.notation()`.  You should obey magnitude changes as a result of `.notation()` not not `.scale()`.

Reply:
Done.